### PR TITLE
Implement new resources - bond,bridge,vlan

### PR DESCRIPTION
**Pull request by robertvarjasi**
Implements new resources: bond,bridge and vlan network interfaces. Docs also generated. Addressing issue #65 too.

### CODE REVIEW

File: maas/resource_maas_block_device.go [hunk 1]
```diff
@@ -386,7 +386,7 @@ func updateBlockDevicePartitions(client *client.Client, d *schema.ResourceData,
 				return err
 			}
 		}
-		if fsType := partition["fs_type"].(string); fsType != "" {
+		if fsType := partition["fs_type"].(string); fsType != "" && fsType != "raid" {
```
Comment (cpg1111) [inline maas/resource_maas_block_device.go:389]: this would be preferable in a separate PR, and explaining why this is needed?

File: maas/resource_maas_network_interface_bridge.go [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if networkInterface == nil {
+		networkInterface, err = client.NetworkInterfaces.CreateBridge(machine.SystemID, getNetworkInterfaceBridgeParams(d))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+	d.SetId(fmt.Sprintf("%v", networkInterface.ID))
```
Comment (cpg1111) [inline maas/resource_maas_network_interface_bridge.go:156]: `networkInterface.ID` should always be an int, can you change the `fmt.Sprintf()` to `strconv.Itoa()`

**Review by cpg1111 — CHANGES_REQUESTED** [review]
Thanks for the contribution @robertvarjasi! There's some feedback I've left on the PR, most notably, I see you have changes to gomaasclient, but rather than using `replace` can you please submit a PR with your changes to github.com/maas/gomaasclient? Additionally, can you submit the raid related changes in a separate PR as to keep commit history concise around what has been added?

**Comment by robertvarjasi** [comment]
> Thanks for the contribution @robertvarjasi! There's some feedback I've left on the PR, most notably, I see you have changes to gomaasclient, but rather than using `replace` can you please submit a PR with your changes to github.com/maas/gomaasclient? Additionally, can you submit the raid related changes in a separate PR as to keep commit history concise around what has been added?

@cpg1111 thanks for your feedback! I will address these problems in the next few days. And ping you if I am finished.

File: maas/resource_maas_block_device.go (hunk 1 above)
Comment (robertvarjasi) [reply maas/resource_maas_block_device.go:389]: otherwise its try to partitioning my raid block device with "raid" fstype and API request will fail. I will reproduce the error message and paste it here.

Comment (cpg1111) [reply maas/resource_maas_block_device.go:389]: It would still be preferable to have this as a separate PR, land that one first, and rebase this.

File: maas/resource_maas_raid.go [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/maas/gomaasclient/client"
+	"github.com/maas/gomaasclient/entity"
+)
+
+func resourceMaasRaid() *schema.Resource {
```
[the remark this answers is not in the document]
Comment (robertvarjasi) [reply maas/resource_maas_raid.go]: removed.

File: maas/resource_maas_network_interface_bridge.go [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+func findNetworkInterfaceBridge(client *client.Client, machineSystemID string, identifier string) (*entity.NetworkInterface, error) {
+	networkInterfaces, err := client.NetworkInterfaces.Get(machineSystemID)
+	if err != nil {
+		return nil, err
+	}
+	for _, n := range networkInterfaces {
+		if n.Type != "bridge" {
+			continue
+		}
+		if n.Name == identifier || fmt.Sprintf("%v", n.ID) == identifier {
```
[the remark this answers is not in the document]
Comment (robertvarjasi) [reply maas/resource_maas_network_interface_bridge.go]: done

**Comment by robertvarjasi** [comment]
@cpg1111 can you help me with the review please?
